src/fread.c

```diff
@@ -653,7 +654,7 @@ static void parse_double_regular(FieldParseContext *ctx)
     acc = 10*acc + digit;
     ch++;
   }
-  sf = ch - ch0;
+  sf = (uint_fast8_t)(ch - ch0);
```

Comment:
This helped me see that `sf` shouldn't be declared as 8 bits on line 649.  A field longer than 256 digits wouldn't get caught. If the overflow in `sf` left say `sf==5`,  an incorrect numeric would be read.  Perhaps `sf` should be `uint_fast32_t`, and declared here too.  Maybe even `size_t` which is the only guaranteed type to correctly subtract any two pointers.

Reply:
good catch, we should probably add a test for this too...